src/core/thread/network_data_local.cpp

```diff
@@ -199,79 +195,18 @@ ThreadError Local::UpdateRloc(BorderRouterTlv &aBorderRouter)
     return kThreadError_None;
 }
 
-ThreadError Local::Register(const Ip6::Address &aDestination)
+ThreadError Local::SendServerDataNotification(void)
 {
     ThreadError error = kThreadError_None;
-    Coap::Header header;
-    Message *message;
-    Ip6::MessageInfo messageInfo;
-    ThreadNetworkDataTlv threadNetworkDataTlv;
 
     UpdateRloc();
-    mSocket.Open(&HandleUdpReceive, this);
-
-    for (size_t i = 0; i < sizeof(mCoapToken); i++)
-    {
-        mCoapToken[i] = static_cast<uint8_t>(otPlatRandomGet());
-    }
-
-    header.Init();
-    header.SetVersion(1);
-    header.SetType(Coap::Header::kTypeConfirmable);
-    header.SetCode(Coap::Header::kCodePost);
-    header.SetMessageId(++mCoapMessageId);
-    header.SetToken(mCoapToken, sizeof(mCoapToken));
-    header.AppendUriPathOptions(OPENTHREAD_URI_SERVER_DATA);
-    header.AppendContentFormatOption(Coap::Header::kApplicationOctetStream);
-    header.Finalize();
-
-    VerifyOrExit((message = Ip6::Udp::NewMessage(0)) != NULL, error = kThreadError_NoBufs);
-    SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
-    threadNetworkDataTlv.Init();
-    threadNetworkDataTlv.SetLength(mLength);
-    SuccessOrExit(error = message->Append(&threadNetworkDataTlv, sizeof(threadNetworkDataTlv)));
-    SuccessOrExit(error = message->Append(mTlvs, mLength));
-
-    memset(&messageInfo, 0, sizeof(messageInfo));
-    memcpy(&messageInfo.mPeerAddr, &aDestination, sizeof(messageInfo.mPeerAddr));
-    messageInfo.mPeerPort = kCoapUdpPort;
-    SuccessOrExit(error = mSocket.SendTo(*message, messageInfo));
-
-    otLogInfoNetData("Sent network data registration\n");
+    SuccessOrExit(error = NetworkData::SendServerDataNotification(true, mOldRloc));
```

Comment:
> ThreadRloc16Tlv is used to update the Leader when a server's RLOC16 changes.
If no rloc changes, ThreadRloc16Tlv should not be included in my understanding.

However for two separate `netdataregister`, there would be ThreadRloc16Tlv in the second server data notification.